pkg/tsdb/sqleng/sql_engine_test.go

```diff
@@ -9,317 +9,294 @@ import (
-			fixtures[6] = int16Value
-			fixtures[7] = &int16Value
-			fixtures[8] = int8Value
-			fixtures[9] = &int8Value
-			fixtures[10] = float64Value
-			fixtures[11] = &float64Value
-			fixtures[12] = float32Value
-			fixtures[13] = &float32Value
-			fixtures[14] = uintValue
-			fixtures[15] = &uintValue
-			fixtures[16] = uint64Value
-			fixtures[17] = &uint64Value
-			fixtures[18] = uint32Value
-			fixtures[19] = &uint32Value
-			fixtures[20] = uint16Value
-			fixtures[21] = &uint16Value
-			fixtures[22] = uint8Value
-			fixtures[23] = &uint8Value
-
-			var intNilPointer *int
-			var int64NilPointer *int64
-			var int32NilPointer *int32
-			var int16NilPointer *int16
-			var int8NilPointer *int8
-			var float64NilPointer *float64
-			var float32NilPointer *float32
-			var uintNilPointer *uint
-			var uint64NilPointer *uint64
-			var uint32NilPointer *uint32
-			var uint16NilPointer *uint16
-			var uint8NilPointer *uint8
-
-			nilPointerFixtures := make([]interface{}, 12)
-			nilPointerFixtures[0] = intNilPointer
-			nilPointerFixtures[1] = int64NilPointer
-			nilPointerFixtures[2] = int32NilPointer
-			nilPointerFixtures[3] = int16NilPointer
-			nilPointerFixtures[4] = int8NilPointer
-			nilPointerFixtures[5] = float64NilPointer
-			nilPointerFixtures[6] = float32NilPointer
-			nilPointerFixtures[7] = uintNilPointer
-			nilPointerFixtures[8] = uint64NilPointer
-			nilPointerFixtures[9] = uint32NilPointer
-			nilPointerFixtures[10] = uint16NilPointer
-			nilPointerFixtures[11] = uint8NilPointer
-
-			Convey("When converting values to float should return expected value", func() {
-				for _, f := range fixtures {
-					value, _ := ConvertSqlValueColumnToFloat("col", f)
-
-					So(value.Valid, ShouldBeTrue)
-					So(value.Float64, ShouldEqual, null.FloatFrom(1).Float64)
-				}
-			})
-
-			Convey("When converting nil pointer values to float should return expected value", func() {
-				for _, f := range nilPointerFixtures {
-					value, err := ConvertSqlValueColumnToFloat("col", f)
-					So(err, ShouldBeNil)
-
-					So(value.Valid, ShouldBeFalse)
-				}
-			})
-		})
+	t.Run("Given row with nil value columns", func(t *testing.T) {
+		var intNilPointer *int
+		var int64NilPointer *int64
+		var int32NilPointer *int32
+		var int16NilPointer *int16
+		var int8NilPointer *int8
+		var float64NilPointer *float64
+		var float32NilPointer *float32
+		var uintNilPointer *uint
+		var uint64NilPointer *uint64
+		var uint32NilPointer *uint32
+		var uint16NilPointer *uint16
+		var uint8NilPointer *uint8
+
+		fixtures := []interface{}{
+			intNilPointer,
+			int64NilPointer,
+			int32NilPointer,
+			int16NilPointer,
+			int8NilPointer,
+			float64NilPointer,
+			float32NilPointer,
+			uintNilPointer,
+			uint64NilPointer,
+			uint32NilPointer,
+			uint16NilPointer,
+			uint8NilPointer,
+		}
+
+		for _, f := range fixtures {
+			value, err := ConvertSqlValueColumnToFloat("col", f)
+			require.NoError(t, err)
+			require.False(t, value.Valid)
```

Comment:
```suggestion
			t.Run("", func(t *testing.T) {
				value, err := ConvertSqlValueColumnToFloat("col", f)
				require.NoError(t, err)
				assert.False(t, value.Valid)
			}
```

Reply:
Updated so that each fixture/test case is run as a subtest.